lib/cc/engine/analyzers/violation.rb

```diff
@@ -88,6 +95,12 @@ def read_up_path
           )
         end
 
+        def fingerprint
+          digest = Digest::MD5.new
+          digest << "#{current_sexp.file}-#{current_sexp.mass}"
```

Comment:
I think these should be separate additions. There's no benefit in constructing the string and then calling `<<`